source/plugin/configuration/loaders.rst

```diff
@@ -33,58 +43,70 @@ First, let's grab a new :javadoc:`HoconConfigurationLoader` that points to our c
 
     Path potentialFile = getConfigPath();
     ConfigurationLoader<CommentedConfigurationNode> loader =
-      HoconConfigurationLoader.builder().setPath(potentialFile).build();
+            HoconConfigurationLoader.builder().setPath(potentialFile).build();
 
 The loader will also hold a generic type depending what kind of node it will build. These *Configuration Nodes* will be
 discussed in :doc:`a later section <nodes>`.
 
-``ConfigurationLoader``\ s usually hold a builder for you to statically access and create a new instance of the loader of
+``ConfigurationLoader``\s usually hold a builder for you to statically access and create a new instance of the loader of
 your desired type. For a basic configuration, we don't really need to specify anything other than the file we want to
 load from and/or save to, so all we'll do is tell it exactly that, using
-``HoconConfigurationLoader.builder().setPath(path)``. We then tell the builder to build the instance (``build()``) for
-it and store it in a variable.
+:javadoc:`AbstractConfigurationLoader.Builder#setPath(Path) {HoconConfigurationLoader.builder().setPath(path)}`.
+We then tell the builder to build the instance (:javadoc:`HoconConfigurationLoader.Builder#build() {build()}`) for it
+and store it in a variable.
 
-Of course, this isn't the only way to load a file. The builder also has the method ``setURL(URL)``, in case you want
-to load a resource without using a ``Path`` object. Bear in mind that configuration loaders created from an ``URL``
-are read-only as they have no way of writing back data to the URL.
+Of course, this isn't the only way to load a file. The builder also has the method
+:javadoc:`AbstractConfigurationLoader.Builder#setPath(URL) {setURL(url)}`, in case you want
+to load a resource without using a :javadoc:`Path` object. Bear in mind that configuration loaders created from an
+:javadoc:`URL` are read-only as they have no way of writing back data to the URL.
 
 This functionality may be used to bundle default configurations with your plugin jar file and load them as initial
 configuration to be edited by the server administrator (or your plugin itself).
 
+.. note::
+
+    This example uses a ``HoconConfigurationLoader``, which is the recommended approach for Sponge plugins, but
+    you can also use a :javadoc:`YAMLConfigurationLoader` or :javadoc:`GsonConfigurationLoader` for loading legacy
+    configs.
+
+
```

Comment:
Minor: extra line

Reply:
I thought about using extra newlines before headlines to visually separate it in the sources as well.
(But I guess I did it very inconsistently)

For now I will remove all double newlines, but maybe its worth a discussion?